src/jswrap_object.c

```diff
@@ -222,8 +222,20 @@ void jswrap_object_keys_or_property_names_cb(
 
     while (symbols) {
       unsigned int i;
-      for (i=0;i<symbols->symbolCount;i++) {
-        JsVar *name = jsvNewFromString(&symbols->symbolChars[symbols->symbols[i].strOffset]);
+      unsigned char symbolCount = READ_FLASH_UINT8(&symbols->symbolCount);
+      unsigned short strOffset = READ_FLASH_UINT16(&symbols->symbols[i].strOffset);
+      for (i=0;i<symbolCount;i++) {
+#ifndef USE_FLASH_MEMORY
+        JsVar *name = jsvNewFromString(&symbols->symbolChars[strOffset]);
+#else
+        // On the esp8266 the string is in flash, so we have to copy it to RAM first
+        // We can't use flash_strncpy here because it assumes that strings start on a word
```

Comment:
Hmm - that's a good idea - it might make things a lot easier.

I always forget, but I wonder if the endianness is such that instead of strcmp, a simple uint32 compare could be used to compare 4 characters at a time - that'd work well with ESP8266 and would be super fast too.